src/uniwill_keyboard.h

```diff
@@ -504,9 +617,109 @@ static int uw_kbd_bl_init(struct platform_device *dev)
 	} else {
 		// For non-RGB versions
 		// Enable keyboard backlight immediately (should it be disabled)
-		uniwill_write_kbd_bl_enable(1);
+		uniwill_write_kbd_bl_enable(0);
+	}
+
+	return status;
+}
+
+
+
+static ssize_t uw_list_power_modes(struct device *child,
+				 struct device_attribute *attr, char *buffer)
+{
+	ssize_t size = 0;
+	u8 i;
+	for (i = 0; i < power_modes_count; i++) {
+		size += sprintf(buffer + size, "%d\n", i);
+	}
+	return size;
+}
+
+static ssize_t uw_power_mode_show(struct device *child,
+				 struct device_attribute *attr, char *buffer)
+{
+	u8 mode = uniwill_get_power_mode();
+	return sprintf(buffer, "%d", mode);
+}
+
+static ssize_t uw_power_mode_store(struct device *child,
+				   struct device_attribute *attr,
+				   const char *buffer, size_t size)
+{
+
+	switch (buffer[0]) {
+		case '0':
+			uniwill_set_power_mode(0);
+			break;
+		case '1':
+			uniwill_set_power_mode(1);
+			break;
+		case '2':
+			uniwill_set_power_mode(2);
+			break;
+		case 'c':
+			uniwill_cycle_power_mode();
+	}
+
+	return size;
+}
+
+
+// Device attributes used by uw kbd
+struct uw_power_mode_dev_attrs_t {
+	struct device_attribute available_power_modes;
+	struct device_attribute power_mode;
+} uw_power_mode_dev_attrs = {
+	.available_power_modes = __ATTR(available_power_modes, 0444, uw_list_power_modes, NULL),
+	.power_mode = __ATTR(power_mode, 0644, uw_power_mode_show, uw_power_mode_store)
+};
+
+// Device attributes used for uw_power_mode
+static struct attribute *uw_power_mode_attrs[] = {
+	&uw_power_mode_dev_attrs.available_power_modes.attr,
+	&uw_power_mode_dev_attrs.power_mode.attr,
+	NULL
+};
+
+static struct attribute_group uw_power_mode_attr_group = {
+	.name = "uw_power_mode",
+	.attrs = uw_power_mode_attrs
+};
+
+static int uw_power_mode_init(struct platform_device *dev)
+{
+	int status = 0;
+	int uniwill_has_power_mode = false
+		// New names
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1501A1650TI")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1501A2060")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1501I1650TI")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1501I2060")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1701A1650TI")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1701A2060")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1701I1650TI")
+		| dmi_match(DMI_BOARD_NAME, "POLARIS1701I2060")
+
+		// Old names
+		// | dmi_match(DMI_BOARD_NAME, "Polaris15I01")
+		// | dmi_match(DMI_BOARD_NAME, "Polaris17I01")
+		// | dmi_match(DMI_BOARD_NAME, "Polaris15A01")
+		// | dmi_match(DMI_BOARD_NAME, "Polaris1501I2060")
+		// | dmi_match(DMI_BOARD_NAME, "Polaris1701I2060")
+		;
+
+
+	if (!uniwill_has_power_mode) {
+		TUXEDO_INFO("PC without power mode");
+		return 127;
 	}
 
+	// Init sysfs power mode attributes group
```

Comment:
Power modes need to be written before this comment.
Setting the different values depending on the laptop and the power modes it allows.
For that, edit `power_modes` and `power_modes_count` at these lines.
Those variables are defined near the top of the file at: https://github.com/tuxedocomputers/tuxedo-keyboard/pull/105/files#diff-e9ab76a5901bf8345c65cfbdd9fc9d43663ccb17544b1dcef52d141b9b3630c6R70-R75



You can find the addresses I know here:
https://github.com/tuxedocomputers/tuxedo-keyboard/pull/105/files#diff-e9ab76a5901bf8345c65cfbdd9fc9d43663ccb17544b1dcef52d141b9b3630c6R61-R67